### DISCUSSION TITLE
Using arrays with relationships in a form

### DISCUSSION THREAD

**Discussion by mistuk72** [Budibase/budibase#13290] [Help]
I'm trying to build an app that allows me to track train journeys. I'm using the inbuilt BudibaseDB.

I have a station_table that has a column with all the station names in the UK, there are other columns too with info such as postcode, lat, long etc.

I have created a journeys table that has a stationStart and stationEnd column - this uses a relationship so that I can choose the stations from the stationName column in the station_table

The bit I'm having difficulty doing is that I want to be able to record the stations I transited between the stationStart and stationEnd. I'm guessing I need to use an array - but I'm not sure what to do here. Clearly each journey will have a different list of stations, and that list will be of different lengths

Also what I'd like to be able to do, when entering the journey information from a form, is to be able to look up the station names for the transit station in the same way that I can for the stationStart and stationEnd.

Can anyone help?

Thanks

**Comment by ConorWebb96** [comment]
Hey @mistuk72,

Relating to what you said below. You could add an additional table that records were a train has traveled on a specific day? After each trip you could use an automation create/update automations to write/update records to this table. How you'd store this would be up to you an array inside a json column would make the most sense.
> I want to be able to record the stations I transited between the stationStart and stationEnd.

You can use a data provider with a [picker](https://docs.budibase.com/docs/option-picker#using-a-data-provider) nested inside it. I believe that this would allow you to search through your table that has your station names. Alternatively, you could use the [type-a-head ](https://github.com/ConorWebb96/bb-type-ahead/)custom plugin. This would allow you to query your stations table and assign a relevant value to this field. (Good for performance.)
> is to be able to look up the station names for the transit station

Hopefully this helps point you in the right direction!
